articles/android-viewpager2/index.md

```diff
@@ -0,0 +1,214 @@
+      holder.textView.text = words[position]
+}
+
+override fun getItemCount(): Int = words.size
+```
+
+In the `onCreateViewHolder` method we inflate the `page_layout` layout using a `LayoutInflater`. In the `onBindViewHolder` method, we assign a word from the word list to the `TextView`. The `getItemCount` is a one line method where we return the size of the word list. That's all for the adapter.
+
+### Step 4 — Testing the ViewPager2 component
+Open `MainActivity.kt` file. In the `onCreate` method, add a list of words and a reference to the `ViewPager2` component.
+
+```Kotlin
+val words = arrayListOf("One", "Two", "Three", "Four", "Five")
+val pager = findViewById<ViewPager2>(R.id.pager)
+```
+
+Then add the adapter as shown.
+
+```Kotlin
+pager.adapter = PagerAdapter(this, words)
+```
+
+That's what we need for the `ViewPager2`. Now build and run the application.
+
+The output should be like to this.
+
+![View pager gif](/engineering-education/android-viewpager2/view-pager.gif)
+
+To change the scroll orientation to vertical, add the following statement.
+
+```Kotlin
+pager.orientation = ViewPager2.ORIENTATION_VERTICAL
+```
+
+It should resemble the demo below.
+
+![vertical orientation gif](/engineering-education/android-viewpager2/vertical-scroll.gif)
+
+### Step 5 — Integrating TabLayout
+Now that our view pager is functioning as expected. Let's go ahead and integrate it with `TabLayout`.
+
+First add the reference to the `TabLayout` component.
+
+```Kotlin
+val tabLayout = findViewById<TabLayout>(R.id.tabLayout)
+```
+
+To integrate `ViewPager2` with `TabLayout` we need to use a `TabLayoutMediator` class. This was quite easier with the old ViewPager. One had to use the `TabLayout`'s `setUpWithViewPager` method and pass the `ViewPager2` refference.
+
+The `TabLayoutMediator` class takes in two parameters, the `TabLayout` and `ViewPager2` references.
+
+```Kotlin
+TabLayoutMediator(tabLayout, pager)
+```
+
+The class then takes a lambda function with two parameters. `TabLayout.Tab` and an integer representing the position the ViewPagers' pages.
+
+```Kotlin
+TabLayoutMediator(tabLayout, pager) {tab, position ->
+      tab.text = "${position + 1}"
+}.attach()
+```
+
+In the function, we set the tabs' text to the position plus one. We add one because the positions start from zero. We attach the components by calling the `attach` method.
+
+Build and run the application. This is how it should look.
+
+![tab layout gif](/engineering-education/android-viewpager2/tab-layout.gif)
+
+Another important feature of `TabLayout` is the on tab selected listener. This notifies the listeners whenever a tab's selected, unselected or reselected. This is useful when one wants to perform some background tasks when the listener fires up. We will use toasts in our application for demonstration.
+
+Add the following code to implement the feature.
+
+```Kotlin
+tabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
```

Comment:
I think Kotlin 1.4 removed the redundant body options in lambda functions. You can check on that and see if it makes this code simpler. But I am not exactly sure. Just do a small research. It's not a big deal, though.